sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/CatalystTypeConvertersSuite.scala

```diff
@@ -223,4 +223,32 @@ class CatalystTypeConvertersSuite extends SparkFunSuite with SQLHelper {
       }
     }
   }
+
+  test("SPARK-34605: converting java.time.Duration to DayTimeIntervalType") {
+    Seq(
+      "PT20.123456S",
+      "P2DT3H4M",
+      "P2D").foreach { intervalStr =>
```

Comment:
I think many people are not familiar with this format. Can we add comments or use a different way to create `Duration`?

Reply:
Cam we add some negative cases?

Reply:
@cloud-fan I created durations using another `Duration` methods. Also added negative tests.